pkg/crdutil/crdutil.go

```diff
@@ -49,26 +51,32 @@ func (s *StringList) Set(value string) error {
 }
 
 var (
-	crdsDir StringList
+	files     []string
+	recursive bool
 )
 
 func initFlags() {
-	flag.Var(&crdsDir, "crds-dir", "Path to the directory containing the CRD manifests")
-	flag.Parse()
-
-	if len(crdsDir) == 0 {
-		log.Fatalf("CRDs directory is required")
+	pflag.CommandLine.AddGoFlagSet(flag.CommandLine)
+	pflag.StringSliceVarP(&files, "filename", "f", files,
+		"The files that contain the configurations to apply.")
```

Comment:
This usage string does not seem to accurately represent the intent of the argument(s). From the PR description, I assume that each `filename` represents a path. Each of these can be a directory or a YAML file path. For directories, the YAML files in the directory are processed.

Reply:
The flag names, the logic and also the description is copied from `kubectl apply` :)

Reply:
But to explain this a bit more:

- the old behavior was indeed recursive by default, but without the possibility to decide this
- the new behavior only handles yaml files in the same directory, if a directory is passed
- with setting the recursive flag all directories will be handled recursively

Let me add tests for this to make this pattern clear